yarn-project/pxe/src/pxe_service/error_enriching.ts

```diff
@@ -0,0 +1,81 @@
+import { type SimulationError, isNoirCallStackUnresolved } from '@aztec/circuit-types';
+import { AztecAddress, Fr, FunctionSelector, PUBLIC_DISPATCH_SELECTOR } from '@aztec/circuits.js';
+import { type DebugLogger } from '@aztec/foundation/log';
+import { resolveAssertionMessage, resolveOpcodeLocations } from '@aztec/simulator';
+
+import { type ContractDataOracle, type PxeDatabase } from '../index.js';
+
+/**
+ * Adds contract and function names to a simulation error.
+ * @param err - The error to enrich.
+ */
```

Comment:
I found it a bit hard to understand what this does based on this description. Does it mutate `err` by doing some further processing on it? If so it'd be good to at least vaguely describe said mutation.

Reply:
I just saw that this was moved from a different file and not introduced here - was confused by this since the scope of this PR is sorta vague. At any rate my comments below still apply.

Reply:
Improved comments and PR description, touching on everything that has been modified (sorta got out of hand, but the end result I think is nice)